### ISSUE TITLE
Branch or option for Vuetify + NUXT template?

### ISSUE THREAD

**Issue by jokeronaldo** [SimulatedGREG/electron-vue#270]
Can you provide an easy installation for vuetify + NUXT?

**Comment by SimulatedGREG** [comment]
@jokeronaldo 

Installing `vuefity` is as simple as `import`ing it and then applying with `Vue.use()`. After that you just need to include needed CSS files. Doing so would provide the same setup as `vuetifyjs/webpack-advanced`, with the exception of its initial landing page components. As for electron-vue providing an option for `vuetify`, this would be a separation of concern and should be up to the developer to install.

> If you are wanting to use Vuetify in an existing project, you can install by using npm or yarn. Keep in mind, you will still need to include Vuetify, Roboto font library, and Material icon library css files as they are not part of the javascript import.<br>
[**Existing applications**](https://vuetifyjs.com/vuetify/quick-start)

As for NUXT, how would using SSR benefit an `electron` application? Or are you asking that electron-vue also produce a SSR web output?

**Comment by jokeronaldo** [comment]
That's the point, I will try to output the same application for desktop and web, so, I don't want to rewrite the same code. Is it possible?

**Comment by SimulatedGREG** [comment]
The current electron-vue boilerplate already supports a minimal [`web`](https://simulatedgreg.gitbooks.io/electron-vue/content/en/webpack-configurations.html#electron-vuewebpackwebconfigjs) output with `npm run build:web`, but as far supporting SSR with Nuxt.js, that won't be happening. Both electron-vue and Nuxt.js do a great job at supporting the extremes of their target platforms, so adding them together into one boilerplate would be a mess.

Even if the time was taken to support both targets in one boilerplate, the developer experience would be awful. You would run into so many anti-patterns and conditionals to test whether you should be hydrating local state, maintaining sessions, using native `electron` APIs, using `<router-link/>` vs `<nuxt-link/>`...the list goes on. In my opinion, *build for all* boilerplates don't usually simplify complex applications.

Now there is potential for creating an electron build with `nuxt generate`, which creates a static non-SSR version of your Nuxt.js application, but because `nuxt` is handling everything there would need to be heavy modifications to optimize it for `electron`, especially when needing to support native node modules and `electron`'s `main` process.

In conclusion, there are very good reasons why electron-vue will not support this type of output.

**Comment by jokeronaldo** [comment]
I'm trying to figure it out how do I share components in repository, may it could save a lot of time, but, the problem is electron file structure.

Ty anyway +1

**Comment by SimulatedGREG** [comment]
No problem.

**Comment by vesper8** [comment]
@jokeronaldo I think a lot of people are in the same position as you, I know I am right now. I would love to reuse as much as possible of my web app's code in using it with Electron

9 months later.. I would love to hear if you came up with a good balanced solution.. or did you abandon either the web or electron version as a result of this challenge?

**Comment by jokeronaldo** [comment]
@vesper8 I left behind it, I'm writing different code for native and web application. Sorry if I can't help you.

**Comment by iplanwebsites** [comment]
While I agree it could be a mess, I'm also looking for patterns that'd facilitate to re-using some code between those platforms. 
@vesper8, Any luck?
